src/libaktualizr/aktualizr_api.h

```diff
@@ -0,0 +1,298 @@
+updates may make a vehicle undrivable for an extended period of time.
+
+Functional Requirements
+=======================
+
+1. Integration with 3rd party HMI
+2. Integration with 3rd party interfaces to install software on secondary ECUs
+3. Limit network traffic and installation to specific vehicle states
+4. Progress bars
+
+Non-functional Requirements
+===========================
+
+1. Since in-vehicle interfaces are often proprietary and under NDA, the
+   implementation must be kept separate from the \aktualizr tree.
+2. It should be possible to have the integration performed by a separate team
+3. Flexibility and simple integration into new environments.
+
+Overview
+========
+
+\aktualizr performs all its operations in a background thread, and all the
+operations except aktualizr_init() and aktualizr_free() are asynchronous.
+Notifications occur via callbacks. These calls are made synchronously from
+\aktualizr's background thread, so they should complete quickly. Normally the
+callback will only post a notification the application's main event loop.
+
+
+Configuration
+=============
+A configuration object is assembled by is built by calling
+aktualizr_config_init() then loading configuration files or directories with
+aktualizr_config_load(). After all the configuration options are loaded, the
+configuration object is passed to aktualizr_init().
+
+The configuration cannot be changed after it has been passed to
+aktualizr_init()
+
+
+Why not IPC?
+============
+
+It would be possible to allow other components to interface with \aktualizr via
+a IPC interface such as D-Bus or Sockets. This would have the some advantages,
+such as:
+
+ - Process Isolation
+ - Easier to assemble pieces together and hack together things with scripts.
+ - Easier privilege separation
+
+However we believe the disadvantages outweigh them:
+
+- Have to build an IPC mechanism
+- Have to build bridges to connect this 'common' IPC mechanism to whatever the
+  end-user system has picked
+- New features may require simultaneous changes to the aktualizr IPC these
+  bridges.
+- Hard to replace built-in functionality with external functionality (e.g.
+  Package Managers)
+- Harder to add steps to the high-level workflow.
+
+It is still possible to write a wrapper that would expose libaktualizr via an
+IPC mechanism, but this can be written with a specific use-case and IPC
+technology in mind.
+
+*/
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+struct aktualizr_struct;
+
+/**
+ * \aktualizr handle
+ */
+typedef aktualizr_struct* aktualizr_t;
+
+struct aktualizr_config_struct;
+
+/** \aktualizr configuration object handle */
+typedef aktualizr_config_struct* aktualizr_config_t;
+
+/**
+ * \aktualizr API result code.
+ */
+typedef enum {
+  /** The operation completed successfully */
+  AKTUALIZR_OK = 0,
+  /** The operation cannot be started because an existing operation is in progress */
+  AKTUALIZR_ALREADY_IN_PROGRESS,
```

Comment:
As this is an external api, setting all enum values explicitly would be more reassuring.